src/scipp/typing.py

```diff
@@ -54,7 +59,7 @@ def has_numeric_type(obj: _std_typing.Any) -> bool:
 A value of ``None`` indicates "all dimensions."
 """
 
-VariableLike = _std_typing.Union[Variable, DataArray, Dataset]
+VariableLike = _std_typing.Union[Variable, DataArray, Dataset, 'DataGroup']
```

Comment:
Will we be able to support `DataGroup` for all operands where `VariableLike` worked, or will we need a new subcategory?

Reply:
I think `VariableLike` is not well defined. It will not work in every case, e.g. there are no `values`, etc. properties. But this is also the case for Dataset. And most code assumes that `shape` returns a tuple of ints, so it will blow up if there is a `None`. Did I miss anything else?